### PR TITLE
🚸 OSQPOptimizer -> OSQP.Optimizer

### PR DESCRIPTION
**Pull request by blegat**

### CODE REVIEW

File: src/MOIWrapper.jl [hunk 1]
```diff
@@ -88,9 +88,16 @@ mutable struct OSQPOptimizer <: MOI.AbstractOptimizer
     end
 end
 
-hasresults(optimizer::OSQPOptimizer) = optimizer.hasresults
+# used to smooth out transition of OSQP v0.4 -> v0.5, TODO: remove on OSQP v0.6
+export OSQPOptimizer
+function OSQPOptimizer()
+    Base.depwarn("OSQPOptimizer() is deprecated, use OSQP.Optimizer() instead.",
+                 :OSQPOptimizer)
```
Comment (mlubin) [inline src/MOIWrapper.jl]: Shouldn't this function return `Optimizer()` to avoid breaking existing code?

Comment (tkoolen) [reply src/MOIWrapper.jl]: It does actually:

```julia
julia> using OSQP.MathOptInterfaceOSQP
[ Info: Precompiling OSQP [ab2f91bb-94b4-55e3-9ba0-7f65df51de79]

julia> optimizer = OSQPOptimizer()
┌ Warning: OSQPOptimizer() is deprecated, use OSQP.Optimizer() instead.
│   caller = top-level scope at none:0
└ @ Core none:0
Optimizer(OSQP.Model(Ptr{OSQP.Workspace} @0x0000000000000000, Float64[], Float64[]), false, OSQP.Results(Float64[], Float64[], OSQP.Info(140496452123152, #undef, 569348, 3, 1.5e-323, 0.0, 0.0, 0.0, 6.94145239783165e-310, 6.94144482996333e-310, 6.9414448298977e-310, 0, 0.0), Float64[], Float64[]), true, Dict{Symbol,Any}(), MinSense::OptimizationSense = 0, 0.0, Float64[], OSQP.MathOptInterfaceOSQP.ModificationCaches.ProblemModificationCache{Float64}(#undef, #undef, #undef, #undef, #undef), OSQP.MathOptInterfaceOSQP.ModificationCaches.WarmStartCache{Float64}(#undef, #undef), Dict{Int64,UnitRange{Int64}}())
```

Note that this is actually exactly equivalent to

```julia
Base.@deprecate OSQPOptimizer() Optimizer()
```

No need to change this in this PR, just FYI.

Comment (tkoolen) [reply src/MOIWrapper.jl]: Ah, I hadn't noticed b52ad84.

Comment (blegat) [reply src/MOIWrapper.jl]: Forgot about this macro, @tkoolen thanks for the tip :)

**Pull request merged**